code/data_structures/src/linked_list/linked_list.c

```diff
@@ -0,0 +1,378 @@
+#ifndef _LINKED_LIST_C_
+#define _LINKED_LIST_C_
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h> // C99 required
+
+typedef unsigned int UINT;
+
+typedef bool (*COMPAREFUNC)(void* pData1, void* pData2);
+typedef void(*TRAVERSEFUNC)(void *pData);
+typedef void(*DESTROYFUNC)(void* pData);
+
+typedef struct LinkedListNode
+{
+    struct LinkedListNode* pNext;
+    void* pData;
+}LinkedListNode, *pLinkedListNode;
+
+typedef struct LinkedList
+{
+    LinkedListNode* pHead;
+    LinkedListNode* pTail;
+    LinkedListNode* pCur;
+    UINT uCount;
+}LinkedList, *pLinkedList;
+
+
+LinkedList* LinkedList_Create();
+void LinkedList_Destroy(LinkedList* pLinkedList, DESTROYFUNC DestroyFunc);
```

Comment:
1. You need to follow our C style guide; Function  names should use snake_case (Fro the functional parts e.g `GetHead` -> `get_head`).
2. Other C style guide issues are also present, such as invalid placing of braces, return types going on their own line, etc. Please check the guide.